test/boost/cdc_test.cc

```diff
@@ -23,14 +23,94 @@
 #include <string>
 
 #include "cdc/log.hh"
+#include "schema_builder.hh"
 #include "test/lib/cql_assertions.hh"
 #include "test/lib/cql_test_env.hh"
 #include "transport/messages/result_message.hh"
+
 #include "types.hh"
 #include "types/tuple.hh"
+#include "types/map.hh"
+#include "types/user.hh"
 
 using namespace std::string_literals;
 
+namespace cdc {
+api::timestamp_type find_timestamp(const schema&, const mutation&);
+}
+
+SEASTAR_THREAD_TEST_CASE(test_find_mutation_timestamp) {
+    do_with_cql_env_thread([] (cql_test_env& e) {
+        cquery_nofail(e, "CREATE TYPE ut (a int, b int)");
+        cquery_nofail(e, "CREATE TABLE ks.t (pk int, ck int, vstatic int static, vint int, "
+                "vmap map<int, int>, vfmap frozen<map<int, int>>, vut ut, vfut frozen<ut>, primary key (pk, ck))");
+
+        auto schema = schema_builder("ks", "t")
+            .with_column("pk", int32_type, column_kind::partition_key)
+            .with_column("vstatic", int32_type, column_kind::static_column)
+            .with_column("ck", int32_type, column_kind::clustering_key)
+            .with_column("vint", int32_type)
+            .with_column("vmap", map_type_impl::get_instance(int32_type, int32_type, true))
+            .with_column("vfmap", map_type_impl::get_instance(int32_type, int32_type, false))
+            .with_column("vut", user_type_impl::get_instance("ks", "ut", {to_bytes("a"), to_bytes("b")}, {int32_type, int32_type}, true))
+            .with_column("vfut", user_type_impl::get_instance("ks", "ut", {to_bytes("a"), to_bytes("b")}, {int32_type, int32_type}, false))
+            .build();
+
+        auto check_stmt = [&] (const sstring& query) {
+            auto muts = e.get_modification_mutations(query).get0();
+            BOOST_REQUIRE(!muts.empty());
+
+            for (auto& m: muts) {
+                auto ts = cdc::find_timestamp(*schema, m);
+                BOOST_REQUIRE(
+                    std::chrono::duration_cast<std::chrono::milliseconds>(
+                        api::timestamp_clock::duration(api::new_timestamp() - ts))
+                    < std::chrono::milliseconds(10));
```

Comment:
I'm trying to test whether "find_timestamp" truly returns the mutation's timestamp. But I don't know the mutation's timestamp (that's what I'm trying to test), so I'm assuming that the mutation's timestamp is close to the current time when I do the check, because the mutation was generated very recently.

It shouldn't be a problem if I set it to 1000 milliseconds, it does make the test a bit less specific, e.g. what if find_timestamp  returns some different timestamp than the mutation's timestamp but it so happens that it's within the 10 second window? But that obviously won't happen if we look at the function's implementation.